### PR TITLE
Implement CognitiveComplexityBuilder with macro-expansion filtering and tests

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Updated to implement CognitiveComplexityBuilder with macro-expansion filtering, along with unit and BDD tests, feature file, and documentation updates. Exposed public API for lint driver integration and future HIR walker wiring. Includes end-to-end test coverage and a design/exec plan reflecting the approach.

## Changes
- Implement cognitive_complexity.rs containing CognitiveComplexityBuilder
  - Fields: score, nesting_stack (Vec<bool> where true = from expansion), effective_depth
  - API: new, record_structural_increment, record_nesting_increment, record_fundamental_increment, push_nesting, pop_nesting, effective_depth, score, build
  - Behavior:
    - record_structural_increment(is_from_expansion): increments score only if !is_from_expansion
    - record_nesting_increment(is_from_expansion): increments score by current effective_depth if !is_from_expansion
    - record_fundamental_increment(is_from_expansion): increments score if !is_from_expansion
    - push_nesting(is_from_expansion): push onto nesting_stack; increment effective_depth only if !is_from_expansion
    - pop_nesting(): pop from nesting_stack; decrement effective_depth only if popped level was not from expansion; panics on empty stack
    - build(): panics if nesting_stack not empty; returns final score
  - Default implemented via new()
- Add unit tests: common/src/brain_type_metrics/cognitive_complexity_tests.rs
  - Cover structural, nesting, fundamental increments (with and without expansion filtering)
  - Test nesting stack behavior (push/pop, depth accounting, mixed expansion levels)
  - Test composite scenarios and edge cases (unbalanced stack, empty stack panic, etc.)
- Add BDD tests and feature: common/src/brain_type_metrics/cognitive_complexity_tests.rs (tests) and common/tests/cognitive_complexity_behaviour.rs plus common/tests/features/cognitive_complexity.feature
  - End-to-end scenarios for empty function, simple/nested patterns, macro-expanded cases, and mixed real/expansion increments
- Wire exports
  - common/src/brain_type_metrics/mod.rs: add pub mod cognitive_complexity;
  - common/src/lib.rs: re-export CognitiveComplexityBuilder alongside existing items
- Documentation updates
  - docs/brain-trust-lints-design.md: insert Implementation decisions (6.2.3) detailing macro-expansion skip strategy, placement of CognitiveComplexityBuilder, nesting depth handling, API surface, and build() balance check
  - docs/execplans/6-2-3-exclude-macro-expanded-spans.md: add the ExecPlan explaining goals, approach, and validation steps
  - docs/roadmap.md: flip 6.2.3 checkbox to done
- Testing plan
  - Unit tests and BDD tests cover all increment types and macro-expansion scenarios
  - Run: cargo test -p common

## Why this change
- Macro-expanded nodes should not contribute to CC to avoid inflating scores from generated code (Clippy issue #14417 context).
- The approach provides a self-contained CognitiveComplexityBuilder decoupled from rustc_private, ready for future integration via a dedicated HIR walker.
- Tracking nesting with a per-level boolean stack enables accurate effective_depth computation, ensuring macro nesting doesn’t inflate subsequent nesting penalties.
- Exposing the API by increment category keeps the builder decoupled from specific HIR node types.

## Acceptance criteria
- CognitiveComplexityBuilder is exported and usable by the lint driver
- Unit tests and BDD tests cover all increment types and macro-expansion scenarios
- Documentation reflects the design and decisions
- Roadmap item 6.2.3 updated to done

## Risks and mitigations
- Panics on unbalanced push/pop: intentional programming contract check; covered by tests
- Macro-expansion behavior relies on correct is_from_expansion input from the HIR walker; future wiring will provide this

## Related artifacts
- docs/execplans/6-2-3-exclude-macro-expanded-spans.md
- docs/brain-trust-lints-design.md
- docs/roadmap.md

📎 **Task**: https://www.devboxer.com/task/de17e021-ef39-4184-b59e-8dd1c6881498

### CODE REVIEW

File: installer/tests/behaviour_cli.rs [hunk 1]
```diff
@@ -302,41 +311,28 @@ fn then_suite_library_is_staged(cli_world: &CliWorld) {
...
-        .expect("staging directory should exist")
-        .filter_map(|e| e.ok())
-        .map(|e| e.file_name().to_string_lossy().to_string())
-        .collect::<Vec<_>>();
+    // When prebuilt artefacts are available the installer downloads them to the
+    // canonical prebuilt path instead of staging to --target-dir.
+    if let Some(dir) = expected_prebuilt_target_dir(channel)
+        && count_matching_files(&PathBuf::from(&dir), &needle) == 1
+    {
+        return;
```
[the remark this answers is not in the document]
Comment (leynos) [reply installer/tests/behaviour_cli.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
